bot/helpers/legacy_uploader.py

```
@@ -117,18 +136,88 @@ async def playlist_upload(metadata, user):
+    # Preserve legacy base path usage for relative computations
+    base_path = f"{Config.DOWNLOAD_BASE_DIR}/{user['r_id']}/"
+
+    def _compute_relative(p: str, base: str | None) -> str:
+        try:
+            p_abs = os.path.abspath(p)
+            if base:
+                base_abs = os.path.abspath(base)
+                if p_abs.startswith(base_abs):
+                    return os.path.relpath(p_abs, base_abs)
+        except Exception:
+            pass
```

Ruff (0.12.2)

`try`-`except`-`pass` detected, consider logging the exception

(S110)